src/server/game/AI/CoreAI/UnitAI.cpp

```diff
@@ -340,5 +340,8 @@ bool NonTankTargetSelector::operator()(Unit const* target) const
     if (_playerOnly && target->GetTypeId() != TYPEID_PLAYER)
         return false;
 
-    return target != _source->GetVictim();
+    if (HostileReference* currentVictim = _source->getThreatManager().getCurrentVictim())
```

Comment:
NPC's don't necessarily have a victim (or victim) when we are selecting a target, like sindragosa when she's flying away but it could be that the npc is casting another spell on someone else, so I thought it would be better to check who is the tank based on threat